src/testing/advection_explicit_periodic.cpp

```diff
@@ -10,180 +10,903 @@
 #include <deal.II/grid/grid_tools.h>
 #include <deal.II/grid/grid_out.h>
 #include <deal.II/grid/grid_in.h>
+#include <deal.II/base/convergence_table.h>
 
 #include "parameters/all_parameters.h"
 #include "parameters/parameters.h"
+//#include "numerical_flux/numerical_flux.h"
 #include "physics/physics_factory.h"
 #include "physics/physics.h"
+#include "dg/dg.h"
 #include "dg/dg_factory.hpp"
 #include "ode_solver/ode_solver_factory.h"
 #include "advection_explicit_periodic.h"
 
 #include<fenv.h>
 
+#include <deal.II/grid/manifold_lib.h>
+#include <deal.II/fe/mapping_q.h>
+#include <fstream>
+
 namespace PHiLiP {
 namespace Tests {
 template <int dim, int nstate>
 AdvectionPeriodic<dim, nstate>::AdvectionPeriodic(const PHiLiP::Parameters::AllParameters *const parameters_input)
 :
 TestsBase::TestsBase(parameters_input)
+, mpi_communicator(MPI_COMM_WORLD)
```

Comment:
`TestBase` contains an `mpi_communicator` and `pcout` already, why not use those instead of creating/initializing it again in the derived class? You can do `this->mpi_communicator` and `this->pcout`. This comment applies to the other `TestBase` derived classes you've introduced in this PR

Reply:
Fixed by leaving it. For some reason I was getting an error so I left it.